File: frontend/src/types/payment.ts
export type PaymentMethod = 'cash' | 'bank_transfer' | 'upi' | 'cheque';

export interface PaymentRead {
  id: string;
  invoice_id: string;
  amount: string;
  payment_date: string;
  method: PaymentMethod;
  notes: string | null;
  recorded_by: string;
  created_at: string;
}

export interface PaymentCreate {
  amount: string;
  payment_date: string;
  method: PaymentMethod;
  notes?: string;
  confirm_duplicate?: boolean;
}

export interface PaymentCreateResponse {
  payment: PaymentRead;
  duplicate_warning: boolean;
  duplicate_of: string | null;
}

File: backend/app/schemas/payment.py
import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.payment import PaymentMethod


class PaymentCreate(BaseModel):
    amount: Decimal
    payment_date: date
    method: PaymentMethod
    notes: str | None = None
    confirm_duplicate: bool = False


class PaymentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    invoice_id: uuid.UUID
    amount: Decimal
    payment_date: date
    method: PaymentMethod
    notes: str | None
    recorded_by: uuid.UUID
    created_at: datetime


class PaymentCreateResponse(BaseModel):
    payment: PaymentRead
    duplicate_warning: bool = False
    duplicate_of: uuid.UUID | None = None
